vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -74,61 +78,87 @@ func computeETag(data []byte) string {
 	return fmt.Sprintf("\"%X\"", sha512.Sum512(data))
 }
 
+func (c *cache) Get() ([]byte, string, error) {
+	c.once.Do(func() {
+		c.bytes, c.etag, c.err = c.BuildCache()
+	})
+	return c.bytes, c.etag, c.err
+}
+
 // NewOpenAPIService builds an OpenAPIService starting with the given spec.
 func NewOpenAPIService(spec *spec.Swagger) (*OpenAPIService, error) {
 	o := &OpenAPIService{}
+	// TODO(DangerOnTheRanger): is it safe to remove this UpdateSpec call?
 	if err := o.UpdateSpec(spec); err != nil {
 		return nil, err
 	}
 	return o, nil
 }
 
 func (o *OpenAPIService) getSwaggerBytes() ([]byte, string, time.Time) {
-	o.rwMutex.RLock()
-	defer o.rwMutex.RUnlock()
-	return o.specBytes, o.specBytesETag, o.lastModified
+	specBytes, etag, err := o.jsonCache.Get()
```

Comment:
Is there a potential race from removing the lock on the get?

Reply:
I think you're right, it could be possible that `o.UpdateSpec` gets called at the same time one of the HTTP handlers gets called. I'll add the lock back.